ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/PostgresLedgerDao.scala

```diff
@@ -658,18 +658,41 @@ private class PostgresLedgerDao(
   private val SQL_GET_LEDGER_ENTRIES = SQL(
     "select * from ledger_entries where ledger_offset>={startInclusive} and ledger_offset<{endExclusive} order by ledger_offset asc")
 
-  //TODO we should use paging instead, check if Alpakka can do that?
+  private def paginatingStream[T](
+      startInclusive: Long,
+      endExclusive: Long,
+      pageSize: Int,
+      queryPage: (Long, Long) => Source[T, NotUsed]): Source[T, NotUsed] =
```

Comment:
i think this relies on the assumption that the stream is monotonically growing. if it was not monotonically growing, you could end up with different results with the pagination as opposed to the single query.

if i'm correct, please note this fact in the code.

Reply:
yes it does. I added a note as you requested.